app/src/components/misc/ExtensionContextInvalidationNotice/index.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useState } from "react";
+import { Typography } from "antd";
+import { trackExtensionContextInvalidated } from "modules/analytics/events/misc/extensionContextInvalidation";
+import "./extensionContextInvalidationNotice.scss";
+
+const ExtensionContextInvalidationNotice: React.FC = () => {
+  const [visible, setVisible] = useState(false);
+
+  useEffect(() => {
+    window.addEventListener("message", (event) => {
+      if (event.data?.event === "requestly:extensionUpdated") {
+        setVisible(true);
+        trackExtensionContextInvalidated(event.data.oldVersion, event.data.newVersion);
```

Comment:
Why didn't you use `PSMH.addMessageListener` here?

Reply:
Done.